packages/maskbook/src/extension/background-script/EthereumServices/createMetaMaskProvider.ts

```diff
@@ -0,0 +1,31 @@
+import { MetaMaskInpageProvider } from '@metamask/inpage-provider'
+import PortStream from 'extension-port-stream'
+import { detect } from 'detect-browser'
+
+const browser = detect()
+const CHROME_ID = 'nkbihfbeogaeaoehlefnkodbefgpgknn'
+const FIREFOX_ID = 'webextension@metamask.io'
+
+export function createMetaMaskProvider(): MetaMaskInpageProvider {
+    let provider
+    try {
+        const currentMetaMaskId = getMetaMaskId()
+        const metamaskPort = chrome.runtime.connect(currentMetaMaskId)
```

Comment:
Please use `browser.runtime.connect`.

iOS support `runtime.connect` but does not support cross-extension connect, please disable it in iOS

Reply:
ok

Reply:
Since iOS does not support Metamask, this snippet of code would not be called on iOS?

Now I just change this to: 

```ts
// ...
const metamaskPort = globalThis.browser?.runtime.connect(currentMetaMaskId)
```